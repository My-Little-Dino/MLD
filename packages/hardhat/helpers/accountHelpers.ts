import { fromBech32 } from '@harmony-js/crypto';
import { isBech32Address } from '@harmony-js/utils';

export const parseOneAddress = (address: string): string => (isBech32Address(address) ? fromBech32(address) : address);
