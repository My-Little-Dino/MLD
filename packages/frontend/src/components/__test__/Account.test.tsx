import React from 'react';
import ReactDOM from 'react-dom';
import { render } from '@testing-library/react';

import { useWeb3React } from '@web3-react/core';
import Account from '../Account';

import { mockComponent } from '../../helpers/testHelper';

jest.mock('@web3-react/core');

jest.mock('../SignOut', () => (props: any) => mockComponent('Wallets', props));
jest.mock('../Wallets', () => (props: any) => mockComponent('Wallets', props));

describe('Test Account component', () => {
	const component = <Account />;

	const setup = () => render(component);

	beforeEach(() => {
		(useWeb3React as jest.Mock).mockReturnValue({
			account: '0xcfffa6ace98eb3464f77af3059389f8f2c18b6e9',
			library: {
				messenger: {
					chainType: 'hmy',
				},
			},
		});
	});

	afterEach(jest.clearAllMocks);

	it('Should render component without crash', () => {
		const div = document.createElement('div');
		div.id = 'root';
		ReactDOM.render(component, div);
		ReactDOM.unmountComponentAtNode(div);
	});

	it('Should match snapshot', () => {
		const { container } = setup();
		expect(container).toMatchSnapshot();
	});
});
