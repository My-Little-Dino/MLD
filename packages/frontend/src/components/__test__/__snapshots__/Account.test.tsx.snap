// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Account component Should match snapshot 1`] = `
<div>
  <div
    class="sc-bdnxRM bJHeMn"
  >
    <span>
      one1el
      ...
      gcvy
    </span>
  </div>
</div>
`;
